pinot-common/src/main/java/org/apache/pinot/common/broker/BrokerSelector.java

```diff
@@ -28,6 +29,10 @@ public interface BrokerSelector {
    */
   String selectBroker(String... tableNames);
 
+  default BrokerInfo selectBrokerInfo(String... tableName) {
+   throw new UnsupportedOperationException("selectBrokerInfo has not been implemented.");
```

Comment:
are we sure we want to have this default implementation? In case someone extends this class, it may be better to receive a compile time error instead of a runtime in production if the main code calls that method and their implementation does not implement it.

Reply:
> If I dont add the default implementation, then compilation will fail.

That is my point. Is this method required? In other words, is it going to be called by the runtime? In that case an implementation of this interface that do not implement the new method will fail at runtime. In that case it is better to fail at compile time instead.

In our case, `PinotQueryResource` will always call this method. Therefore we cannot inject there old implementations of `BrokerSelector` that do not implement this method.

> Also it is not required that implementations use BrokerInfo. They can continue to use strings.

That is a problem. Either implementations have implement this method or this method should not be in this interface. The alternative implies to create a new interface (that may extend `BrokerSelector`) and `PinotQueryResource` should inject this new interface instead.

IMHO the easiest solution is to do not define this as default and fail at compilation time if the implementation of `BrokerSelector` does not implement this new method

Reply:
I've added a `BrokerInfoSelector` and the derived classes use `BrokerInfo`. Other selectors like `SimpleBrokerSelector` implement `BrokerSelector` and do not change. `PinotQueryResource` uses `BrokerInfoSelector`